### DISCUSSION TITLE
Error on first start. TT

### DISCUSSION THREAD

**Discussion by DevJIP** [kriasoft/react-starter-kit#2020] [Q&A] [answered]
I just cloned and executed "yarn run start".
Can i get some advice?

 yarn run start

  VITE v3.2.4  ready in 690 ms

  ➜  Local:   http://127.0.0.1:5173/
  ➜  Network: use --host to expose
X [ERROR] Could not read from file: D:/Development/Project/MyProject/frontEnd/.yarn/__virtual__/@mui-material-virtual-4d8e2e947b/6/C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/@mui-material-npm-5.10.14-8d1f74b3b3-8.zip/node_modules/@mui/material/esm/styles/index.js

X [ERROR] Could not read from file: D:/Development/Project/MyProject/frontEnd/.yarn/__virtual__/@mui-icons-material-virtual-76ce175ae4/6/C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/@mui-icons-material-npm-5.10.14-021b4f0747-8.zip/node_modules/@mui/icons-material/esm/index.js

X [ERROR] Could not read from file: D:/Development/Project/MyProject/frontEnd/.yarn/__virtual__/react-dom-virtual-22fafa95af/6/C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/react-dom-npm-18.2.0-dd675bca1c-8.zip/node_modules/react-dom/client.js

X [ERROR] Could not read from file: D:/Development/Project/MyProject/frontEnd/.yarn/__virtual__/recoil-virtual-44fa142702/6/C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/recoil-npm-0.7.6-6f8de04e3d-8.zip/node_modules/recoil/es/index.js

X [ERROR] Could not read from file: D:/Development/Project/MyProject/frontEnd/.yarn/__virtual__/react-router-dom-virtual-bf15a87266/6/C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/react-router-dom-npm-6.4.3-3b0e4f134b-8.zip/node_modules/react-router-dom/dist/index.js

X [ERROR] Could not read from file: D:/Development/Project/MyProject/frontEnd/.yarn/__virtual__/@mui-material-virtual-4d8e2e947b/6/C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/@mui-material-npm-5.10.14-8d1f74b3b3-8.zip/node_modules/@mui/material/esm/index.js

X [ERROR] Could not read from file: D:\Development\Project\MyProject\frontEnd\.yarn\__virtual__\@firebase-auth-virtual-4f84b2703f\6\C:\Users\MyComputer\AppData\Local\Yarn\Berry\cache\@firebase-auth-npm-0.20.11-4272f19deb-8.zip\node_modules\@firebase\auth\dist\esm2017\index.js

    C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/firebase-npm-9.14.0-f6f56921c3-8.zip/node_modules/firebase/auth/dist/index.esm.js:1:14:
      1 │ export * from '@firebase/auth';
        ╵               ~~~~~~~~~~~~~~~~

X [ERROR] Could not read from file: D:\Development\Project\MyProject\frontEnd\.yarn\__virtual__\@firebase-analytics-virtual-08607154d6\6\C:\Users\MyComputer\AppData\Local\Yarn\Berry\cache\@firebase-analytics-npm-0.8.4-62ae89307a-8.zip\node_modules\@firebase\analytics\dist\esm\index.esm2017.js

    C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/firebase-npm-9.14.0-f6f56921c3-8.zip/node_modules/firebase/analytics/dist/index.esm.js:1:14:
      1 │ export * from '@firebase/analytics';
        ╵               ~~~~~~~~~~~~~~~~~~~~~

D:\Development\Project\MyProject\frontEnd\.yarn\unplugged\esbuild-npm-0.15.14-0f40bc75be\node_modules\esbuild\lib\main.js:1566
  let error = new Error(`${text}${summary}`);
              ^

Error: Build failed with 8 errors:
error: Could not read from file: D:/Development/Project/MyProject/frontEnd/.yarn/__virtual__/@mui-material-virtual-4d8e2e947b/6/C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/@mui-material-npm-5.10.14-8d1f74b3b3-8.zip/node_modules/@mui/material/esm/styles/index.js
error: Could not read from file: D:/Development/Project/MyProject/frontEnd/.yarn/__virtual__/@mui-icons-material-virtual-76ce175ae4/6/C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/@mui-icons-material-npm-5.10.14-021b4f0747-8.zip/node_modules/@mui/icons-material/esm/index.js
error: Could not read from file: D:/Development/Project/MyProject/frontEnd/.yarn/__virtual__/react-dom-virtual-22fafa95af/6/C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/react-dom-npm-18.2.0-dd675bca1c-8.zip/node_modules/react-dom/client.js
error: Could not read from file: D:/Development/Project/MyProject/frontEnd/.yarn/__virtual__/recoil-virtual-44fa142702/6/C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/recoil-npm-0.7.6-6f8de04e3d-8.zip/node_modules/recoil/es/index.js      
error: Could not read from file: D:/Development/Project/MyProject/frontEnd/.yarn/__virtual__/react-router-dom-virtual-bf15a87266/6/C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/react-router-dom-npm-6.4.3-3b0e4f134b-8.zip/node_modules/react-router-dom/dist/index.js
...
    at failureErrorWithLog (D:\Development\Project\MyProject\frontEnd\.yarn\unplugged\esbuild-npm-0.15.14-0f40bc75be\node_modules\esbuild\lib\main.js:1566:15)
    at D:\Development\Project\MyProject\frontEnd\.yarn\unplugged\esbuild-npm-0.15.14-0f40bc75be\node_modules\esbuild\lib\main.js:1024:28
    at runOnEndCallbacks (D:\Development\Project\MyProject\frontEnd\.yarn\unplugged\esbuild-npm-0.15.14-0f40bc75be\node_modules\esbuild\lib\main.js:1438:61)
    at buildResponseToResult (D:\Development\Project\MyProject\frontEnd\.yarn\unplugged\esbuild-npm-0.15.14-0f40bc75be\node_modules\esbuild\lib\main.js:1022:7)
    at D:\Development\Project\MyProject\frontEnd\.yarn\unplugged\esbuild-npm-0.15.14-0f40bc75be\node_modules\esbuild\lib\main.js:1134:14
    at responseCallbacks.<computed> (D:\Development\Project\MyProject\frontEnd\.yarn\unplugged\esbuild-npm-0.15.14-0f40bc75be\node_modules\esbuild\lib\main.js:671:9)
    at handleIncomingPacket (D:\Development\Project\MyProject\frontEnd\.yarn\unplugged\esbuild-npm-0.15.14-0f40bc75be\node_modules\esbuild\lib\main.js:726:9)
    at Socket.readFromStdout (D:\Development\Project\MyProject\frontEnd\.yarn\unplugged\esbuild-npm-0.15.14-0f40bc75be\node_modules\esbuild\lib\main.js:647:7)
    at Socket.emit (node:events:527:28)
    at addChunk (node:internal/streams/readable:315:12) {
  errors: [
    {
      detail: undefined,
      id: '',
      location: null,
      notes: [],
      pluginName: '',
      text: 'Could not read from file: D:/Development/Project/MyProject/frontEnd/.yarn/__virtual__/@mui-material-virtual-4d8e2e947b/6/C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/@mui-material-npm-5.10.14-8d1f74b3b3-8.zip/node_modules/@mui/material/esm/styles/index.js'
    },
    {
      detail: undefined,
      id: '',
      location: null,
      notes: [],
      pluginName: '',
      text: 'Could not read from file: D:/Development/Project/MyProject/frontEnd/.yarn/__virtual__/@mui-icons-material-virtual-76ce175ae4/6/C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/@mui-icons-material-npm-5.10.14-021b4f0747-8.zip/node_modules/@mui/icons-material/esm/index.js'
    },
    {
      detail: undefined,
      id: '',
      location: null,
      notes: [],
      pluginName: '',
      text: 'Could not read from file: D:/Development/Project/MyProject/frontEnd/.yarn/__virtual__/react-dom-virtual-22fafa95af/6/C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/react-dom-npm-18.2.0-dd675bca1c-8.zip/node_modules/react-dom/client.js'
    },
    {
      detail: undefined,
      id: '',
      location: null,
      notes: [],
      pluginName: '',
      text: 'Could not read from file: D:/Development/Project/MyProject/frontEnd/.yarn/__virtual__/recoil-virtual-44fa142702/6/C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/recoil-npm-0.7.6-6f8de04e3d-8.zip/node_modules/recoil/es/index.js'
    },
    {
      detail: undefined,
      id: '',
      location: null,
      notes: [],
      pluginName: '',
      text: 'Could not read from file: D:/Development/Project/MyProject/frontEnd/.yarn/__virtual__/react-router-dom-virtual-bf15a87266/6/C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/react-router-dom-npm-6.4.3-3b0e4f134b-8.zip/node_modules/react-router-dom/dist/index.js'
    },
    {
      detail: undefined,
      id: '',
      location: null,
      notes: [],
      pluginName: '',
      text: 'Could not read from file: D:/Development/Project/MyProject/frontEnd/.yarn/__virtual__/@mui-material-virtual-4d8e2e947b/6/C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/@mui-material-npm-5.10.14-8d1f74b3b3-8.zip/node_modules/@mui/material/esm/index.js'
    },
    {
      detail: undefined,
      id: '',
      location: {
        column: 14,
        file: 'C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/firebase-npm-9.14.0-f6f56921c3-8.zip/node_modules/firebase/analytics/dist/index.esm.js',
        length: 21,
        line: 1,
        lineText: "export * from '@firebase/analytics';",
        namespace: '',
        suggestion: ''
      },
      notes: [],
      pluginName: '',
      text: 'Could not read from file: D:\\Development\\Project\\MyProject\\frontEnd\\.yarn\\__virtual__\\@firebase-analytics-virtual-08607154d6\\6\\C:\\Users\\MyComputer\\AppData\\Local\\Yarn\\Berry\\cache\\@firebase-analytics-npm-0.8.4-62ae89307a-8.zip\\node_modules\\@firebase\\analytics\\dist\\esm\\index.esm2017.js'
    },
    {
      detail: undefined,
      id: '',
      location: {
        column: 14,
        file: 'C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/firebase-npm-9.14.0-f6f56921c3-8.zip/node_modules/firebase/auth/dist/index.esm.js',
        length: 16,
        line: 1,
        lineText: "export * from '@firebase/auth';",
        namespace: '',
        suggestion: ''
      },
      notes: [],
      pluginName: '',
      text: 'Could not read from file: D:\\Development\\Project\\MyProject\\frontEnd\\.yarn\\__virtual__\\@firebase-auth-virtual-4f84b2703f\\6\\C:\\Users\\MyComputer\\AppData\\Local\\Yarn\\Berry\\cache\\@firebase-auth-npm-0.20.11-4272f19deb-8.zip\\node_modules\\@firebase\\auth\\dist\\esm2017\\index.js'
    }
  ],
  warnings: []
}
Error: Command failed with exit code 1: yarn workspace app run start
    at makeError (file:///C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/execa-npm-6.1.0-be1d7f323b-8.zip/node_modules/execa/lib/error.js:59:11)
    at handlePromise (file:///C:/Users/MyComputer/AppData/Local/Yarn/Berry/cache/execa-npm-6.1.0-be1d7f323b-8.zip/node_modules/execa/index.js:119:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async file:///D:/Development/Project/MyProject/frontEnd/scripts/start.js:6:1 {
  shortMessage: 'Command failed with exit code 1: yarn workspace app run start',
  command: 'yarn workspace app run start',
  escapedCommand: 'yarn workspace app run start',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: undefined,
  stderr: undefined,
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}

**Comment by fullerzz** [comment]
Following as I'm running into the same issue.

**Comment by yuvrajbajwa10** [comment, marked as answer]
Hi guys, I got this directly from discord. 
koistya — 17/12/2022 01:49
@sawhighla there is an issue with ESBuild that Vite is using internally in dev mode that doesn't know how to load (Yarn) dependencies from different drives (e.g. when global cache is on C:/ but the project folder is on E:/.

You can disable global cache to prevent this from happening:

```
yarn config set enableGlobalCache false
```

https://github.com/kriasoft/react-starter-kit/blob/main/.yarnrc.yml#L1
